src/varcache.c

```diff
@@ -79,12 +132,37 @@ bool varcache_set(VarCache *cache, const char *key, const char *value)
 	return true;
 }
 
+bool varcache_set_quoted(PgSocket *client, const char *key, const char *value)
+{
+	char qbuf[400];
+	const char *tmp;
+
+	tmp = value;
+	/* Only quote search_path. Quoting other parameters such as client_encoding
+	 * does not work well some clients like psycopg.*/
+	if (strcmp("search_path", key) == 0) {
+
+		if (!pg_quote_literal(qbuf, value, sizeof(qbuf))) {
+			slog_warning(client, "could not quote parameter: %s=%s", key, value);
```

Comment:
is quoting search_path correct?

These are 2 different things:
```sql
SET search_path TO public, private;
SET search_path TO "public, private";
```

The second one refers to a schema created using `CREATE SCHEMA "public, private"`

Reply:
Emel is adding more tests for the quoting now. I also think the new code is incorrect in some ways. One thing to note as that `search_path` is special compared to the previous items in varcache because it's a GUC_LIST. So these two lines are equivalent:
```sql
SET search_path TO public, private;
SET search_path TO "public", "private";
```

Reply:
Nvm, this function is dead code now. So it can definitely be removed.